src/reduction/alpha.rs

```diff
@@ -75,12 +75,15 @@ pub fn reduced_alpha_channel(png: &PngImage, optimize_alpha: bool) -> Option<Png
     };
 
     let mut raw_data = Vec::with_capacity(png.data.len());
-    for pixel in png.data.chunks(bpp) {
+    for pixel in png.data.chunks_exact(bpp) {
         match transparency_pixel {
             Some(trns) if pixel.iter().skip(colored_bytes).all(|b| *b == 0) => {
                 raw_data.resize(raw_data.len() + colored_bytes, trns);
             }
-            _ => raw_data.extend_from_slice(&pixel[0..colored_bytes]),
+            _ if colored_bytes == 6 => raw_data.extend_from_slice(&pixel[0..6]),
+            _ if colored_bytes == 3 => raw_data.extend_from_slice(&pixel[0..3]),
+            _ if colored_bytes == 2 => raw_data.extend_from_slice(&pixel[0..2]),
+            _ => raw_data.push(pixel[0]),
```

Comment:
This does end up 30-40% faster in the benchmark, but is there a more generic way to improve performance here?

Reply:
This reminds me of [performance issues with how Rust emits `memcpy` calls that I've stumbled upon before](https://github.com/rust-lang/rust/issues/92993). What I learned there is that as application programmers we only can come up with these specific constructions to convince the compiler to emit more efficient code, which is at odds with the generality we usually want, so the answer to your question is probably not.

However, I think adding a comment stating why you made specialized branches for a few colored bytes count would help with code maintenance.

Reply:
Sounds reasonable, feel free to visit this when you want to!